hyperspy/drawing/_markers/horizontal_line_segment.py

```diff
@@ -64,6 +64,10 @@ def __init__(self, x1, x2, y, **kwargs):
         self.marker_properties = lp
         self.set_data(x1=x1, x2=x2, y1=y)
         self.set_marker_properties(**kwargs)
+        # for compatibility with matplotlib 3.1.x - 3.4.x
+        if 'zorder' in self.marker_properties:
+            if self.marker_properties['zorder'] is None:
+                del self.marker_properties['zorder']
```

Comment:
This behaviour can be confusing for the user as it would remove the `zorder` without warning or error for this specific marker. Instead, I would suggest the following:
```suggestion
        if (Version(maptlotlib.__version__) <= Version("3.4")
            and "zorder" in kwargs.keys()):
            _logger.warning("`zorder` kwargs is only supported "
                            "with matplotlib >= 3.5.")
        self.set_marker_properties(**kwargs)
```

This would have the advantage of not hiding anything to the user and to allow using "zorder" if matplotlib supports it.

The other alternative is to do nothing and ignore if the `zorder` is passed and matplotlib doesn't support it. In this case, matplotlib should raise an error or a warning - I haven't check if it does so, but this is what I would expect!

Reply:
The zorder property itself is suppoted on matplotlib 3.1,x, but the behaviour of zorder=None is different for some objects.
I already removed the default value of zorder=None, so, I can remove these lines.
(In such case, we need to write version dependent code on users script like you shows)